app/helpers/application_helper/toolbar_builder.rb

```diff
@@ -1126,11 +1128,19 @@ def build_toolbar_disable_button(id)
         return N_("Read Only Domain cannot be deleted.") unless editable_domain
       when "miq_ae_domain_edit"
         return N_("Read Only Domain cannot be edited") unless editable_domain
-      when "miq_ae_domain_lock"
+      when "miq_ae_domain_lock", "miq_ae_namespace_edit"
         return N_("Domain is Locked.") unless editable_domain
       when "miq_ae_domain_unlock"
         return N_("Domain is Unlocked.") if editable_domain
       end
+    when "MiqAeNamespace", "MiqAeClass", "MiqAeInstance", "MiqAeMethod"
+      if %w(miq_ae_namespace_copy miq_ae_instance_copy miq_ae_class_copy miq_ae_method_copy).include?(id) &&
+        !User.current_tenant.any_editable_domains? &&
+        !MiqAeDomain.any_unlocked? &&
+        !editable_domain?(@record)
+
+        return N_("No writeable domains available, create one first")
```

Comment:
@hayesr how about we change hover text for the button to "At least one domain should be marked as editable & unlocked"